common/src/main/java/io/pravega/common/util/CopyOnWriteMapUtils.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utils class that contains copy-on-write methods for maps.
+ */
+public final class CopyOnWriteMapUtils {
```

Comment:
Rather than externalizing the logic, I would rather have it internalized to a single class. For example can we Make a `CopyOnWriteHashMap` which internally does the copies? Because this would allow callers to not need to understand the details of the implementation.

Reply:
Added `CopyOnWriteHashMap` according to your comment.